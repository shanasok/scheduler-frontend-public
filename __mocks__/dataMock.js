export const lessons = [
    {
        name: "Mock Threading the Machine",
        lessonNumber: 1,
    },
    {
        name: "Mock Winding a Bobbin",
        lessonNumber: 1,
    }
];

export const courses =
    {
        data: {
            allCourses: [
                {
                    id: 1,
                    name: "testCourseName",
                    maxAttendees: 5,
                    summary: "testCourseSummary",
                    subjects: [
                        {
                            id: 1,
                            name: "testSubjectName",
                            summary: "testSubjectSummary",
                            subjectNumber: 1,
                            lessons: [
                                {
                                    id: 1,
                                    name: "Mock Threading the Machine",
                                    lessonNumber: 1,
                                    pages: [],
                                },
                                {
                                    id: 2,
                                    name: "Mock Winding a Bobbin",
                                    lessonNumber: 2,
                                    pages: [],
                                }
                            ]
                        }
                    ]
                }
            ]
        }
    }

export const courses2 =
    {
        "data": {
            "allCourses": [
                {
                    "id": 1,
                    "name": "testCourseName",
                    "maxAttendees": 5,
                    "summary": "testCourseSummary",
                    "subjects": [
                        {
                            "id": 1,
                            "name": "testSubjectName",
                            "summary": "testSubjectSummary",
                            "subjectNumber": 1,
                            "lessons": []
                        }
                    ]
                }
            ]
        }
    }

export const lessonContentData = {
    "data": {
        "findLessonById": {
            "id": 4,
            "name": "Introduction",
            "lessonNumber": 1,
            "pages": [
                {
                    "id": 1,
                    "pageNumber": 1,
                    "pageContent": [
                        {
                            "contentNumber": 4,
                            "content": {
                                "id": 4,
                                "contentType": "DEFINITION",
                                "term": "JVM",
                                "explanation": "Java Virtual Machine"
                            }
                        },
                        {
                            "contentNumber": 1,
                            "content": {
                                "id": 1,
                                "contentType": "FACT",
                                "fact": "Java is an object-oriented programming language"
                            }
                        },
                        {
                            "contentNumber": 2,
                            "content": {
                                "id": 2,
                                "contentType": "FACT",
                                "fact": "Java is platform independent"
                            }
                        },
                        {
                            "contentNumber": 3,
                            "content": {
                                "id": 3,
                                "contentType": "FACT",
                                "fact": "Compiler converts Java code to Java bytecode"
                            }
                        },
                        {
                            "contentNumber": 5,
                            "content": {
                                "id": 5,
                                "contentType": "FACT",
                                "fact": "JVM executes Java bytecode"
                            }
                        }
                    ]
                },
                {
                    "id": 2,
                    "pageNumber": 2,
                    "pageContent": []
                },
                {
                    "id": 3,
                    "pageNumber": 3,
                    "pageContent": []
                }
            ]
        }
    }
}