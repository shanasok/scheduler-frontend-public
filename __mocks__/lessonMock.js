export const courses =  [
    {
        "id": 1,
        "name": "Sewing",
        "maxAttendees": 5,
        "summary": "Learn how to use a sewing machine and sew garments and items for your home.",
        "subjects": [
            {
                "id": 1,
                "name": "SEWING BASICS",
                "summary": "Learn to thread and use a machine. Create simple projects using straight and zig-zag stitches.",
                "subjectNumber": 1,
                "lessons": [
                    {
                        "id": 1,
                        "name": "Threading the machine",
                        "lessonNumber": 1,
                        "pages": []
                    },
                    {
                        "id": 2,
                        "name": "Sewing a zip",
                        "lessonNumber": 2,
                        "pages": []
                    }
                ]
            },
            {
                "id": 2,
                "name": "DRESS MAKING BASICS",
                "summary": "Learn to make simple garments from patterns.",
                "subjectNumber": 2,
                "lessons": [
                    {
                        "id": 3,
                        "name": "Hand sew a hem",
                        "lessonNumber": 3,
                        "pages": []
                    }
                ]
            }
        ]
    },
    {
        "id": 2,
        "name": "Java",
        "maxAttendees": 12,
        "summary": "Learn to code in Java using good Object Oriented design principles such as SOLID.",
        "subjects": [
            {
                "id": 6,
                "name": "Java Basics",
                "summary": "Learn the basics of object oriented programming from variables and classes to for loops and methods. ",
                "subjectNumber": 1,
                "lessons": [
                    {
                        "id": 4,
                        "name": "Introduction",
                        "lessonNumber": 1,
                        "pages": [
                            {
                                "id": 1,
                                "pageNumber": 1
                            },
                            {
                                "id": 2,
                                "pageNumber": 2
                            },
                            {
                                "id": 3,
                                "pageNumber": 3
                            }
                        ]
                    },
                    {
                        "id": 5,
                        "name": "Primitive Data Types",
                        "lessonNumber": 2,
                        "pages": []
                    },
                    {
                        "id": 6,
                        "name": "Objects",
                        "lessonNumber": 3,
                        "pages": []
                    },
                    {
                        "id": 7,
                        "name": "Constructors",
                        "lessonNumber": 4,
                        "pages": []
                    },
                    {
                        "id": 8,
                        "name": "Loops",
                        "lessonNumber": 5,
                        "pages": []
                    },
                    {
                        "id": 9,
                        "name": "Modifiers",
                        "lessonNumber": 6,
                        "pages": []
                    },
                    {
                        "id": 10,
                        "name": "Enums",
                        "lessonNumber": 7,
                        "pages": []
                    },
                    {
                        "id": 11,
                        "name": "Collections",
                        "lessonNumber": 8,
                        "pages": []
                    }
                ]
            }
        ]
    },
    {
        "id": 3,
        "name": "Web Development",
        "maxAttendees": 12,
        "summary": "Learn to code web pages using technologies such as CSS, HTML and JavaScript.",
        "subjects": [
            {
                "id": 3,
                "name": "HTML",
                "summary": "Learn HTML and start putting content in your web pages.",
                "subjectNumber": 1,
                "lessons": []
            },
            {
                "id": 4,
                "name": "CSS",
                "summary": "Learn the basics of CSS and use it to style your web pages.",
                "subjectNumber": 2,
                "lessons": []
            },
            {
                "id": 5,
                "name": "JavaScript",
                "summary": "Use JavaScript to make your, web pages responsive.",
                "subjectNumber": 3,
                "lessons": []
            }
        ]
    }
];

export const lessonContent = {
    "id": 4,
    "name": "Introduction",
    "lessonNumber": 1,
    "pages": [
        {
            "id": 1,
            "pageNumber": 1,
            "pageContent": [
                {
                    "contentNumber": 4,
                    "isEditable": false,
                    "content": {
                        "id": 4,
                        "contentType": "DEFINITION",
                        "term": "JVM",
                        "explanation": "Java Virtual Machine"
                    }
                },
                {
                    "contentNumber": 1,
                    "isEditable": false,
                    "content": {
                        "id": 1,
                        "contentType": "FACT",
                        "fact": "Java is an object-oriented programming language"
                    }
                },
                {
                    "contentNumber": 2,
                    "isEditable": false,
                    "content": {
                        "id": 2,
                        "contentType": "FACT",
                        "fact": "Java is platform independent"
                    }
                },
                {
                    "contentNumber": 3,
                    "isEditable": false,
                    "content": {
                        "id": 3,
                        "contentType": "FACT",
                        "fact": "Compiler converts Java code to Java bytecode"
                    }
                },
                {
                    "contentNumber": 5,
                    "isEditable": false,
                    "content": {
                        "id": 5,
                        "contentType": "FACT",
                        "fact": "JVM executes Java bytecode"
                    }
                }
            ]
        },
        {
            "id": 2,
            "pageNumber": 2,
            "pageContent": []
        },
        {
            "id": 3,
            "pageNumber": 3,
            "pageContent": []
        }
    ]
};

