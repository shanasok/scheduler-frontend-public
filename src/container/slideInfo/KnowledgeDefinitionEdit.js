import React from 'react';
import {bindActionCreators, compose} from "redux";
import {withRouter} from "react-router-dom";
import {connect} from "react-redux";
import {setCurrentLessonContent} from "../../redux/actions/lessonContentAction";
import TeacherSavePageContentIcon from "../teacher/TeacherSavePageContentIcon";

class KnowledgePointDefinitionEdit extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            termNewVal: (this.props.content? this.props.content.term : ""),
            explanationNewVal: (this.props.content? this.props.content.explanation : ""),
        };

;       this.updateKnowledgePointDefinitionTermInState = this.updateKnowledgePointDefinitionTermInState.bind(this);
        this.updateKnowledgePointDefinitionExplanationInState = this.updateKnowledgePointDefinitionExplanationInState.bind(this);
    }

    updateKnowledgePointDefinitionTermInState(event) {
        this.setState({termNewVal: event.target.value});
    }

    updateKnowledgePointDefinitionExplanationInState(event) {
        this.setState({explanationNewVal: event.target.value});
    }

    render() {
        const term = (this.props.content)? this.props.content.term : "";
        const explanation = (this.props.content)? this.props.content.explanation : "";

        return <div className="knowledgePoint">
            <TeacherSavePageContentIcon
                contentId={this.props.contentId}
                newValues={{
                    'term':this.state.termNewVal,
                    'explanation':this.state.explanationNewVal,
                }}
            />
            <input defaultValue={term}
                   className="definitionTermEditBox"
                   onChange={(e) => this.updateKnowledgePointDefinitionTermInState(
                       e,
                       this.props.contentId,
                       this.props.currentLessonContent)} /> =
            <input defaultValue={explanation}
                   className="definitionExplanationEditBox"
                   onChange={(e) => this.updateKnowledgePointDefinitionExplanationInState(
                       e,
                       this.props.contentId,
                       this.props.currentLessonContent)} />
        </div>


    }
}

const mapStateToProps = state => ({
    currentLessonContent: state.currentLessonContent,
});

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            setCurrentLessonContent,
        }, dispatch)
    }
}

export default compose(
    withRouter,
    connect(mapStateToProps, mapDispatchToProps)
)(KnowledgePointDefinitionEdit);
