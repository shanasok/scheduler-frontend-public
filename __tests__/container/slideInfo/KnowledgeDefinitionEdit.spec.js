import React from 'react';
import KnowledgeDefinitionEdit from '../../../src/container/slideInfo/KnowledgeDefinitionEdit';
import renderer from 'react-test-renderer';
import configureMockStore from "redux-mock-store";
import {configure, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {Provider as ReduxProvider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import {lessonContent} from "../../../__mocks__/lessonMock";
import {MemoryRouter} from "react-router";

const mockStore = configureMockStore()
const mockState = {
    lesson: lessonContent
}
let store = mockStore(mockState)
configure({adapter: new Adapter()});

it('renders correctly when definition is given', () => {

    const content = {
        contentType: "DEFINITION",
        explanation: "test explanation",
        term: "test term",
        id: "1",
    }

    const tree = renderer
        .create(<BrowserRouter><ReduxProvider store={store}><KnowledgeDefinitionEdit content={content}/></ReduxProvider></BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[1].props['defaultValue']).toEqual(content.term);
    expect(tree.children[3].props['defaultValue']).toEqual(content.explanation);
});

it('renders correctly when definition is not given', () => {

    const tree = renderer
        .create(<BrowserRouter><ReduxProvider store={store}><KnowledgeDefinitionEdit /></ReduxProvider></BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[1].props['defaultValue']).toEqual("");
});

it("should call setState on definition term", () => {
    const content = {
        'term':'testTerm',
        'explanation':'testExplanation',
    }

    const mockEventTerm = {
        target: {
            value: "newTestTerm"
        }
    };

    const mockEventExplanation = {
        target: {
            value: "newTestExplanation"
        }
    };

    const expected = {
        'termNewVal':'newTestTerm',
        'explanationNewVal':'newTestExplanation',
    }

    let wrapper = mount(
        <MemoryRouter initialEntries={[ '/' ]}>
            <ReduxProvider store={store}>
                <KnowledgeDefinitionEdit content={content}/>
            </ReduxProvider>
        </MemoryRouter>
    );

    wrapper.find('.definitionTermEditBox').simulate('change',mockEventTerm);
    wrapper.instance().updateKnowledgePointDefinitionTermInState = jest.fn();
    wrapper.instance().updateKnowledgePointDefinitionTermInState(mockEventTerm);
    expect(wrapper.instance().updateKnowledgePointDefinitionTermInState).toHaveBeenCalledWith(mockEventTerm);


    wrapper.find('.definitionExplanationEditBox').simulate('change',mockEventExplanation);
    wrapper.instance().updateKnowledgePointDefinitionExplanationInState = jest.fn();
    wrapper.instance().updateKnowledgePointDefinitionExplanationInState(mockEventExplanation);
    expect(wrapper.instance().updateKnowledgePointDefinitionExplanationInState).toHaveBeenCalledWith(mockEventExplanation);

    expect(wrapper.childAt(0).childAt(0).childAt(0).childAt(0).childAt(0)
        .instance().state).toEqual(expected);
});