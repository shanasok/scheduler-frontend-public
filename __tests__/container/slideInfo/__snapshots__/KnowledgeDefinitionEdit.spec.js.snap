// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when definition is given 1`] = `
<div
  className="knowledgePoint"
>
  <i
    className="fas fa-save mouseOverBecomeArrow"
    onClick={[Function]}
  />
  <input
    className="definitionTermEditBox"
    defaultValue="test term"
    onChange={[Function]}
  />
   =
  <input
    className="definitionExplanationEditBox"
    defaultValue="test explanation"
    onChange={[Function]}
  />
</div>
`;

exports[`renders correctly when definition is not given 1`] = `
<div
  className="knowledgePoint"
>
  <i
    className="fas fa-save mouseOverBecomeArrow"
    onClick={[Function]}
  />
  <input
    className="definitionTermEditBox"
    defaultValue=""
    onChange={[Function]}
  />
   =
  <input
    className="definitionExplanationEditBox"
    defaultValue=""
    onChange={[Function]}
  />
</div>
`;
